tools/rpkg/src/duckdbr.cpp

```diff
@@ -1169,6 +1170,113 @@ SEXP duckdb_unregister_R(SEXP connsexp, SEXP namesexp) {
 	return R_NilValue;
 }
 
+class RArrowTabularStreamFactory {
+public:
+	RArrowTabularStreamFactory(SEXP export_fun_p, SEXP arrow_tabular_p)
+	    : arrow_tabular(arrow_tabular_p), export_fun(export_fun_p) {};
+	static unique_ptr<ArrowArrayStreamWrapper> Produce(uintptr_t factory_p) {
+		RProtector r;
+		int err;
+		auto factory = (RArrowTabularStreamFactory *)factory_p;
+
+		// export the arrow tabular to a stream pointer
+		auto export_call = r.Protect(Rf_lang2(factory->export_fun, factory->arrow_tabular));
+		SEXP stream_ptr_sexp = r.Protect(R_tryEval(export_call, R_GlobalEnv, &err));
+		if (err) {
+			Rf_error("Failed to produce arrow stream");
+		}
+		// TODO this stream pointer leaks?
+		// TODO this may fail on 32 bit
```

Comment:
Thanks, copied the snippet. I believe you mean the R integration? I don't think tests are run there, but they probably should.

Reply:
No, I mean arrow's pyarrow `reticulate` integration, which uses the C interface too: https://github.com/apache/arrow/blob/master/r/tests/testthat/test-python.R

I'm not certain whether those test runners have Python and run those pyarrow integration tests. If they do, this would cover 32- and 64-bit since the Windows package is biarch.